### PR TITLE
Update blob quickstart code sample

### PR DESCRIPTION
**Pull request by pauljewellmsft**
Snippets from this code sample are used in the [blob quickstart for Go](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-go). Once this code sample looks good, I'll update the quickstart.

### CODE REVIEW

**Comment by pauljewellmsft** [comment]
@sandeep-sen / @jhendrixMSFT

Could either of you please review when you get a chance? This is the code sample we pull the blob quickstart snippets from. Thanks so much!

File: storage-quickstart.go [hunk 1]
```diff
@@ -26,107 +24,88 @@ import (
...
+	handleError(err)
+
+	serviceClient := client.ServiceClient()
 
 	// Create the container
-	containerName := fmt.Sprintf("quickstart-%s", randomString())
+	containerName := "quickstart-sample-container"
 	fmt.Printf("Creating a container named %s\n", containerName)
 	containerClient := serviceClient.NewContainerClient(containerName)
 	_, err = containerClient.Create(ctx, nil)
```
Comment (jhendrixMSFT) [inline storage-quickstart.go:52]: This can be replaced with `client.CreateContainer(...)`

File: storage-quickstart.go [hunk 2]
```diff
@@ -26,107 +24,88 @@ import (
...
+	handleError(err)
 
-	serviceClient, err := azblob.NewServiceClient(url, credential, nil)
-	if err != nil {
-		log.Fatal("Invalid credentials with error: " + err.Error())
-	}
+	client, err := azblob.NewClient(url, credential, nil)
+	handleError(err)
+
+	serviceClient := client.ServiceClient()
```
Comment (jhendrixMSFT) [inline storage-quickstart.go:46]: This can be removed.

File: storage-quickstart.go [hunk 3]
```diff
@@ -26,107 +24,88 @@ import (
...
+	blobClient := containerClient.NewBlockBlobClient(blobName)
 
 	// Upload to data to blob storage
-	_, err = blobClient.UploadBufferToBlockBlob(ctx, data, azblob.HighLevelUploadToBlockBlobOption{})
-
-	if err != nil {
-		log.Fatalf("Failure to upload to blob: %+v", err)
-	}
+	fmt.Printf("Uploading a blob named %s\n", blobName)
+	_, err = blobClient.UploadBuffer(ctx, data, &azblob.UploadBufferOptions{})
```
Comment (jhendrixMSFT) [inline storage-quickstart.go:62]: This can be replaced with `client.UploadBuffer(...)`

File: storage-quickstart.go [hunk 4]
```diff
@@ -26,107 +24,88 @@ import (
...
-	data := []byte("\nhello world this is a blob\n")
-	blobName := "quickstartblob" + "-" + randomString()
+	data := []byte("\nHello, world! This is a blob.\n")
+	blobName := "sample-blob"
 
-	blobClient, err := azblob.NewBlockBlobClient(url+containerName+"/"+blobName, credential, nil)
-	if err != nil {
-		log.Fatal(err)
-	}
+	blobClient := containerClient.NewBlockBlobClient(blobName)
```
Comment (jhendrixMSFT) [inline storage-quickstart.go:58]: This can be removed.

File: storage-quickstart.go [hunk 5]
```diff
@@ -26,107 +24,88 @@ import (
...
-	}
+	fmt.Printf("Uploading a blob named %s\n", blobName)
+	_, err = blobClient.UploadBuffer(ctx, data, &azblob.UploadBufferOptions{})
+	handleError(err)
 
 	// List the blobs in the container
 	fmt.Println("Listing the blobs in the container:")
 
-	pager := containerClient.ListBlobsFlat(nil)
+	pager := containerClient.NewListBlobsFlatPager(&container.ListBlobsFlatOptions{
```
Comment (jhendrixMSFT) [inline storage-quickstart.go:68]: `client.NewListBlobsFlatPager(...)`

File: storage-quickstart.go [hunk 6]
```diff
@@ -26,107 +24,88 @@ import (
...
-	if err = pager.Err(); err != nil {
-		log.Fatalf("Failure to list blobs: %+v", err)
-	}
-
 	// Download the blob
-	get, err := blobClient.Download(ctx, nil)
-	if err != nil {
-		log.Fatal(err)
-	}
+	get, err := blobClient.DownloadStream(ctx, nil)
```
Comment (jhendrixMSFT) [inline storage-quickstart.go:82]: `client.DownloadStream(...)`

File: storage-quickstart.go [hunk 7]
```diff
@@ -26,107 +24,88 @@ import (
...
+	handleError(err)
 
-	downloadedData := &bytes.Buffer{}
-	reader := get.Body(azblob.RetryReaderOptions{})
+	downloadedData := bytes.Buffer{}
+	reader := get.Body
 	_, err = downloadedData.ReadFrom(reader)
```
Comment (jhendrixMSFT) [inline storage-quickstart.go:87]: This is the simple case but has reliability problems (i.e. no retries on an interrupted stream).  Consider switching this out to something like this.
```go
rr := get.NewRetryReader(...)
downloadedData.ReadFrom(rr)
```

File: storage-quickstart.go [hunk 8]
```diff
@@ -26,107 +24,88 @@ import (
...
 
-	fmt.Printf("Press enter key to delete the blob fils, example container, and exit the application.\n")
+	fmt.Printf("Press enter key to delete resources and exit the application.\n")
 	bufio.NewReader(os.Stdin).ReadBytes('\n')
 	fmt.Printf("Cleaning up.\n")
 
 	// Delete the blob
 	fmt.Printf("Deleting the blob " + blobName + "\n")
 
 	_, err = blobClient.Delete(ctx, nil)
```
Comment (jhendrixMSFT) [inline storage-quickstart.go:104]: `client.DeleteBlob(...)`

File: storage-quickstart.go [hunk 9]
```diff
@@ -26,107 +24,88 @@ import (
...
-	if err != nil {
-		log.Fatalf("Failure: %+v", err)
-	}
+	handleError(err)
 
 	// Delete the container
-	fmt.Printf("Deleting the blob " + containerName + "\n")
+	fmt.Printf("Deleting the container " + containerName + "\n")
 	_, err = containerClient.Delete(ctx, nil)
```
Comment (jhendrixMSFT) [inline storage-quickstart.go:109]: `client.DeleteContainer(...)`

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
We want to promote the capabilities of the `azblob.Client` as it should cover the 95% use-case.

**Pull request merged**